internal/client/client.go

```diff
@@ -0,0 +1,419 @@
+// Copyright 2024 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package client provides a rest client to talk to the Trusty API.
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	packageurl "github.com/package-url/packageurl-go"
+	khttp "sigs.k8s.io/release-utils/http"
+
+	v1types "github.com/stacklok/trusty-sdk-go/pkg/v1/types"
+	v2types "github.com/stacklok/trusty-sdk-go/pkg/v2/types"
+)
+
+const (
+	defaultEndpoint = "https://gh.trustypkg.dev"
```

Comment:
Seems trusty is moving away from this one. Let's switch it to api.trustypkg.dev

Reply:
mmm, I might have messed up a rebase or something, thanks for noticing.